server/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java

```diff
@@ -688,48 +688,59 @@ public AnalyzedRelation visitWithQuery(WithQuery node, StatementAnalysisContext
     protected AnalyzedRelation visitTable(Table<?> node, StatementAnalysisContext context) {
         QualifiedName tableQualifiedName = node.getName();
         SearchPath searchPath = context.sessionSettings().searchPath();
-        AnalyzedRelation relation;
         var relationContext = context.currentRelationContext();
 
         var withQuery = relationContext.parentSources()
             .getAncestor(RelationName.of(tableQualifiedName, null));
         if (withQuery != null) {
-            relation = withQuery;
+            relationContext.addSourceRelation(withQuery);
+            return withQuery;
         } else {
             TableInfo tableInfo;
-            try {
-                tableInfo = schemas.resolveTableInfo(
-                    tableQualifiedName,
-                    context.currentOperation(),
-                    context.sessionSettings().sessionUser(),
-                    searchPath
-                );
-                if (tableInfo instanceof DocTableInfo docTable) {
-                    // Dispatching of doc relations is based on the returned class of the schema information.
-                    relation = new DocTableRelation(docTable);
-                } else {
-                    relation = new TableRelation(tableInfo);
-                }
-            } catch (RelationUnknown e) {
-                View view;
+            var searchPathIterator = searchPath.iterator();
+            while (searchPathIterator.hasNext()) {
```

Comment:
Originally searched for tables from all searchPaths then views from all search paths. Now searching tables and views for each searchPaths.